src/client/components/vision/camera/view_model.ts

```diff
@@ -0,0 +1,282 @@
+import { observable } from 'mobx'
+import { computed } from 'mobx'
+import { createTransformer } from 'mobx-utils'
+import { BufferGeometry } from 'three'
+import { PlaneBufferGeometry } from 'three'
+import { RawShaderMaterial } from 'three'
+import { MeshBasicMaterial } from 'three'
+import { WebGLRenderTarget } from 'three'
+import { Scene } from 'three'
+import { Mesh } from 'three'
+import { WebGLRenderer } from 'three'
+import { Camera } from 'three'
+import { Object3D } from 'three'
+import { OrthographicCamera } from 'three'
+import { Material } from 'three'
+import { DataTexture } from 'three'
+import { LuminanceFormat } from 'three'
+import { RGBFormat } from 'three'
+import { Texture } from 'three'
+import { UnsignedByteType } from 'three'
+import { ClampToEdgeWrapping } from 'three'
+import { LinearFilter } from 'three'
+import { NearestFilter } from 'three'
+import { PlaneGeometry } from 'three'
+import { Matrix4 } from 'three'
+import { Vector4 } from 'three'
+import { Vector2 } from 'three'
+import { Geometry } from 'three'
+
+import { memoize } from '../../../base/memoize'
+
+import { CameraModel } from './model'
+import { ImageModel } from './model'
+import * as bayerFragmentShader from './shaders/bayer.frag'
+import * as bayerVertexShader from './shaders/bayer.vert'
+
+export function fourcc(code: string): number {
```

Comment:
What does this do? Either document through comments or tests.

Reply:
👍